File: array/second_largest_element.py
def find_second_largest_element(nums):
	if len(nums) < 2:
		return None

	first = second = float('-inf')

	for num in nums:
		if num > first:
			second = first
			first = num
		elif num > second and num != first:
			second = num

	return second if second != float('-inf') else None



arr = [2,3,1,5,4]
print(find_second_largest_element(arr))

File: array/second_largest_element.js
function secondHighest(s) {
    if (s.length < 2) {
        return -1;
    }
    let first = -Infinity;
    let second = -Infinity;
    for (const c of s) {
        const num = parseInt(c);
        if (Number.isNaN(num)) {
            continue;
        }
        if (num > first) {
            second = first;
            first = num;
        }
        else if (num > second) {
            second = num;
        }
    }
    return second === -Infinity ? -1 : second;
}
console.log(secondHighest("dfa12321afd"));
// console.log('HIiii')
